addons/web/models/models.py

```diff
@@ -616,9 +629,20 @@ def _web_read_group_format(
         for dict_group in result:
             dict_group['__extra_domain'] = AND(dict_group.pop('__extra_domains'))
 
+        monetary_agg_cur_agg_mapping = self._get_mapping_currency_aggregates(aggregates)
+        # Create a mapping between a monetary aggregate and the column of its currency field aggregate
+        currency_agg_column_mapping = {
+            currency_field_agg: [row[aggregates.index(currency_field_agg) + len(groupby)] for row in groups]
+            for currency_field_agg in set(monetary_agg_cur_agg_mapping.values())
```

Comment:
Nitpick: In terms of perf, you may transform `columns_iterator` into `columns = list(zip(*groups))` and index inside that one. 

```suggestion
            currency_field_agg: [row[index] for row in groups]
            for currency_field_agg in set(monetary_agg_cur_agg_mapping.values())
            if (index := aggregates.index(currency_field_agg) + len(groupby))  # compute once
```